crypto/encrypt.go

```diff
@@ -0,0 +1,56 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+)
+
+// EncryptWithAES256 encrypts data using a AES256 cryptography.
+func EncryptWithAES256(secretKey, nonce, data []byte) ([]byte, error) {
+	if len(secretKey) != 32 {
+		return nil, fmt.Errorf("secret key is not for AES-256: total %d bits", 8*len(secretKey))
+	}
+
+	block, err := aes.NewCipher(secretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	aesgcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nonce) != aesgcm.NonceSize() {
+		return nil, fmt.Errorf("nonce length must be %v", aesgcm.NonceSize())
+	}
+
+	cipherText := aesgcm.Seal(nil, nonce, data, nil)
```

Comment:
We need to append nonce like below.
```suggestion
	cipherText := aesgcm.Seal(nonce, nonce, data, nil)

```

Reply:
As I know, we don't need to pass `nonce` as the 1st arg (which is `dst`). We're passing `nonce` as the 2nd arg.

Reply:
Ah. you're saying the way to append the `nonce` in front of the ciphertext (instead of passing the `nonce` separately to the receiver). Then, you're right.